src/deploy/project.py

```diff
@@ -214,6 +214,37 @@ def _get_service(service):
 
         return matched_services
 
+
+    def is_release_deployed(self, release, environment_id):
+        """
+        Checks the release_id matches the `deployment:label` tag on the services
+        and tasks within those services. We check that the desiredCount of tasks 
+        matches the running count of tasks.
+        """
+        ecs_services = self.get_ecs_services(release, environment_id)
+
+        deployed_services = []
+        for _, services in sorted(ecs_services.items()):
+            for service in services:
+                tags = service["response"]["tags"]
+                desired_task_count = service["response"]["desiredCount"]
+                service_deployment_label = next(tag["value"] for tag in tags if tag["key"] == "deployment:label")
```

Comment:
You might want to look at https://github.com/wellcomecollection/weco-deploy/blob/8d254b231c3ecc55c86bde29de76f1456cacdbea/src/deploy/tags.py:

```python
tags = parse_tags(service["response"]["tags"])
service_deployment_label = tags["deployment:label"]
```

Reply:
Thanks, changed